sklearn/datasets/_arff_parser.py

```diff
@@ -364,17 +371,19 @@ def _pandas_arff_parser(
         elif column_dtype.lower() == "nominal":
             dtypes[name] = "category"
 
+    default_read_csv_kwargs = {
+        "header": None,
+        "na_values": ["?"],  # missing values are represented by `?`
+        "comment": "%",  # skip line starting by `%` since they are comments
+        "quotechar": '"',  # delimiter to use for quoted strings
+        "names": [name for name in openml_columns_info],
+        "dtype": dtypes,
+        "skipinitialspace": True,  # skip spaces after delimiter to follow ARFF specs
+    }
+    read_csv_kwargs = {**default_read_csv_kwargs, **(read_csv_kwargs or {})}
+
     # ARFF represents missing values with "?"
-    frame = pd.read_csv(
-        gzip_file,
-        header=None,
-        na_values=["?"],  # missing values are represented by `?`
-        comment="%",  # skip line starting by `%` since they are comments
-        quotechar='"',  # delimiter to use for quoted strings
-        names=[name for name in openml_columns_info],
-        dtype=dtypes,
-        skipinitialspace=True,  # skip spaces after delimiter to follow ARFF specs
-    )
+    frame = pd.read_csv(gzip_file, **read_csv_kwargs)
```

Comment:
Currently, if there is an exception while reading the data, one would need to enter a debugger to find out where the file is and what the `read_csv_kwargs` are. I think it would be helpful reraise an exception that outputs the `read_csv_kwargs` and `gzip_file` to help with debugging the issue.